packages/presentation/lib/main_page.dart

```diff
@@ -0,0 +1,160 @@
+// ignore_for_file: prefer_const_constructors
+
+import 'package:domain/usecase/check_number_usecase.dart';
+import 'package:domain/usecase/generate_number_usecase.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'main_bloc.dart';
+import 'main_event.dart';
+import 'main_state.dart';
+
+class MyApp extends StatefulWidget {
+  const MyApp({Key? key}) : super(key: key);
+
+  @override
+  State<MyApp> createState() => _MyAppState();
+}
+
+class _MyAppState extends State<MyApp> {
+  @override
+  Widget build(BuildContext context) {
+    return BlocProvider<MainBloc>(
+      create: (_) => MainBloc(
+        checkNumberUseCase: CheckNumberUseCase(),
+        generateNumberUseCase: GenerateNumberUsecase(),
+      ),
+      child: MaterialApp(
+        title: 'Flutter Demo',
+        debugShowCheckedModeBanner: false,
+        theme: ThemeData(
+          primarySwatch: Colors.deepPurple,
+        ),
+        home: const MyHomePage(),
+      ),
+    );
+  }
+}
+
+class MyHomePage extends StatelessWidget {
+  const MyHomePage({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text('Guess the Number'),
+        centerTitle: true,
+      ),
+      body: SafeArea(
+        child: Center(
+          child: BlocConsumer<MainBloc, MainState>(
+            listener: (context, state) {
+              if (state is YouWonState) {
+                showDialog(
+                  context: context,
+                  builder: (context) {
+                    return AlertDialog(
+                      content: Text(
+                        (state).result,
+                        style: const TextStyle(fontSize: 30),
+                      ),
+                    );
+                  },
+                );
+              }
+              if (state is WrongState) {
+                showDialog(
+                  context: context,
+                  builder: (context) {
+                    return AlertDialog(
+                      content: Text(
+                        (state).result,
+                        style: const TextStyle(fontSize: 30),
+                      ),
+                    );
+                  },
+                );
+              }
+              if (state is OutOfAttemptsState) {
```

Comment:
(optional) u have the same logic for all states. u can create abstract class ResultState, place here result variable and in this case u can do this:

```
if (state is ResultState) {
showDialog(
                   context: context,
                   builder: (context) {
                     return AlertDialog(
                       content: Text(
                         (state).result,
                         style: const TextStyle(fontSize: 30),
                       ),
                     );
                   },
                 );
}
```